src/harness/fourslash.ts

```diff
@@ -144,10 +144,10 @@ module FourSlash {
             if (globalOptions.hasOwnProperty(prop)) {
                 switch (prop) {
                     case metadataOptionNames.allowNonTsExtensions:
-                        settings.allowNonTsExtensions = true;
+                        settings.allowNonTsExtensions = globalOptions[prop] === "true";

```

Comment:
You can consider just performing this once in a variable prior to the switch, and then using that variable in the appropriate locations.